src/vellum/lib/test_suite.py

```diff
@@ -0,0 +1,175 @@
+import os
+import time
+from typing import Any, Callable, Union
+from uuid import uuid4
+from ..client import Vellum
+from ..types import (
+    ChatMessage,
+    NamedTestCaseJsonVariableValueRequest,
+    NamedTestCaseNumberVariableValueRequest,
+    NamedTestCaseStringVariableValueRequest,
+    SearchResult,
+    TestSuiteRunExecution,
+    TestSuiteRunMetricOutput,
+    TestSuiteRunState,
+    VellumError,
+    TestSuiteRunExternalExecConfigRequest,
+    TestSuiteRunExternalExecConfigDataRequest,
+    ExternalTestCaseExecution,
+)
+
+
+class VellumTestSuiteResults:
+    def __init__(self, id: str) -> None:
+        self.id = id
+        self.client = Vellum(
+            api_key=os.environ.get("VELLUM_API_KEY"),
+        )
+        self.test_suite_run_state = TestSuiteRunState.QUEUED
+        self.test_suite_run_executions: list[TestSuiteRunExecution] | None = None
+
+    def _refresh_test_suite_run_state(self):
+        test_suite_run = self.client.test_suite_runs.retrieve(self.id)
+        self.test_suite_run_state = test_suite_run.state
+
+    def _ensure_metric_results_are_loaded(self):
+        if self.test_suite_run_executions is not None:
+            return
+
+        self._refresh_test_suite_run_state()
+
+        while self.test_suite_run_state in {TestSuiteRunState.QUEUED, TestSuiteRunState.RUNNING}:
+            time.sleep(1)
+            self._refresh_test_suite_run_state()
+            # TODO: Max retries?
+
+        if self.test_suite_run_state == TestSuiteRunState.FAILED:
+            raise Exception("Test suite run failed")
+
+        if self.test_suite_run_state == TestSuiteRunState.CANCELLED:
+            raise Exception("Test suite run was cancelled")
+
+        response = self.client.test_suite_runs.list_executions(
+            self.id, expand=["results.metric_results.metric_definition"]
+        )
+        self.test_suite_run_executions = response.results
+
+    def get_metric_outputs(
+        self, metric_identifier: str | None = None, output_identifier: str | None = None
+    ) -> list[list[TestSuiteRunMetricOutput]]:
+        self._ensure_metric_results_are_loaded()
+
+        all_metric_results = [execution.metric_results for execution in self.test_suite_run_executions]
+
+        filtered_metric_outputs = [
```

Comment:
We should perform validation such that if you provide a metric identifier that's not unique based on `metric_id`, we raise an exception. For example, if you passed in `metric_definition_label`, but that Metric Definition was used twice in the Test Suite, we should raise an exception.

Reply:
do we not want to allow users to be able to say "get me all of my metric results that use exact match"?